### ISSUE TITLE
GHC 8

### ISSUE THREAD

**Issue by o1lo01ol1o** [tensorflow/haskell#38]
Bumping the master stack.yaml to lts-7.3 unsurprisingly causes the build to break in OSX el capitan:

```
[...]
tensorflow-opgen-0.1.0.0: copy/register
Progress: 12/17
--  While building package tensorflow-0.1.0.0 using:
      /Users/.../.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.24.0.0-ghc-8.0.1 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.24.0.0 build lib:tensorflow --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/.../tensorflow-haskell/.stack-work/logs/tensorflow-0.1.0.0.log

    Configuring tensorflow-0.1.0.0...
    Preprocessing library tensorflow-0.1.0.0...
    [ 4 of 12] Compiling TensorFlow.Types ( src/TensorFlow/Types.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.0.0/build/TensorFlow/Types.o )
    
    /Users/.../tensorflow-haskell/tensorflow/src/TensorFlow/Types.hs:376:5: error:
        Malformed head of type or class declaration: as \\ b : bs
```

  Is there any notion of when GHC 8 might be supported?

**Comment by blackgnezdo** [comment]
Do you have a particular requirement forcing you to use GHC 8?

We would like to retain 7.10 compatibility for some time. If you can suggest a fix making both versions work, we'll probably take it.

**Comment by o1lo01ol1o** [comment]
the datahaskell group would like to target ghc 8+ in working toward a consolidated datascience environment in haskell  (still a ways off, but nonetheless .  .  .).    

I'll try to take a look at the problematic bits if I get some time.  

Thanks for the reply.

**Comment by judah** [comment]
Fixed by #97
